app/components/publications/overview/reports/generate-report-modal.js

```diff
@@ -177,55 +178,54 @@ export default class GenerateReportModalComponent extends Component {
-
-    const currentQueryOptions = {
-      ...commonQueryOptions,
-      // HACK: mu-cl-resources quirk: has-no is intended to be used with relationships,
-      //   but seems to be working with an attribute in this case.
-      //   It might change with mu-cl-resource updates.
-      'filter[mandatees][:has-no:end]': true,
-    };
-    const currentMandateePersons = this.store.query(
-      'person',
-      currentQueryOptions
-    );
+    // // As long as mu-cl-resources does not support an OR filter
+    // //    that allows the end date to be empty or after a specific date
+    // //    we need to separate the filter in two requests
+
+    // const commonQueryOptions = {
+    //   'filter[:has:mandatees]': true,
+    //   // DISABLED: query timeouts
+    //   // 'filter[mandatees][mandate][role][:id:]': this.visibleRoles.map((role) => role.id).join(','),
+    //   // active ranges of mandatees are stored as dateTimes, but with time set to 0:00 UTC
+    //   // since the frontend is in a different timezone, we need to compensate for this
+    //   'filter[mandatees][:lt:start]':
+    //     toDateWithoutUTCOffset(dateRangeEnd).toISOString(),
+    //   'filter[last-name]': searchText, // Ember Data leaves this of when set to undefined (=> no filtering)
+    //   // although we sort and paginate again on the frontend
+    //   //   after combining both query results
+    //   //   sorting an pagination are useful for limiting the payload size
+    //   sort: 'last-name',
+    //   'page[size]': CONFIG.PAGE_SIZE.SELECT,
+    // };
+
+    // const pastQueryOptions = {
+    //   ...commonQueryOptions,
+    //   'filter[mandatees][:gte:end]':
+    //     toDateWithoutUTCOffset(dateRangeStart).toISOString(),
+    // };
+    // const pastMandateePersons = this.store.query('person', pastQueryOptions);
+
+    // const currentQueryOptions = {
+    //   ...commonQueryOptions,
+    //   // HACK: mu-cl-resources quirk: has-no is intended to be used with relationships,
+    //   //   but seems to be working with an attribute in this case.
+    //   //   It might change with mu-cl-resource updates.
+    //   'filter[mandatees][:has-no:end]': true,
+    // };
+    // const currentMandateePersons = this.store.query(
+    //   'person',
+    //   currentQueryOptions
+    // );
+
+    const allMandatees = yield this.mandatees.getMandateesActiveForRange.perform(dateRangeStart, dateRangeEnd, searchText);
+    const allPersons= [];
+    for (const mandatee of allMandatees) {
+      const person =yield mandatee.person;
+      allPersons.addObject(person);
+    }
 
-    let mandateePersons = yield Promise.all([
-      pastMandateePersons,
-      currentMandateePersons,
-    ]);
-    mandateePersons = mandateePersons
-      .map((persons) => persons.toArray())
-      .flat()
-      .uniq()
-      .sortBy('lastName')
-      .slice(0, CONFIG.PAGE_SIZE.SELECT);
+    const mandateePersons = allPersons
+      .sortBy('lastName');
```

Comment:
De `uniq()` had hier zijn nut. Een plain array kan je makkelijk uniquifyen door hem in een `Set` te te steken en terug in array te expanden.

Reply:
met deze `allPersons.addObject(person);` ga je al een unieke lijst maken.
Dus de `uniq()` was niet meer nodig.